### ISSUE TITLE
aws_security_group - Multiple ingress blocks mixing issue

### ISSUE THREAD

**Issue by xakraz** [terraform-providers/terraform-provider-aws#4740]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.3
+ provider.aws v1.11.0
```

### Affected Resource(s)

* aws_security_group

### Terraform Configuration Files

```hcl

resource "aws_security_group" "my-sg" {
    name        = "sg_name"
    description = "sg_desc"
    vpc_id      = "xxxx"

    ingress {
        from_port       = 5432
        to_port         = 5432
        protocol        = "tcp"
        security_groups = [
         ...
        ]
        self            = false
    }

     ingress {
        description = "Bastion"
        from_port       = 5432
        to_port         = 5432
        protocol        = "tcp"
        cidr_blocks = "${var.cidr_blocks}"
        self            = false
    }

    egress {
        from_port       = 0
        to_port         = 0
        protocol        = "-1"
        cidr_blocks     = ["0.0.0.0/0"]
    }

    tags {
        "Name" = "sg_name"
    }
}
```

### Expected Behavior

Terraform creates the excpeted rules

### Actual Behavior

Terraform is trying to delete a rules that does not exist, which is the 1st one, reported with a description which is the value of the 2nd one.
Fails when trying to delete since the rule does not exist.

### Steps to Reproduce

1. Write a TF file for an `aws_security_group` with 2 `ingress` blocks, 1 with a description, the other without
2. `terraform plan && terraform apply`
3. It says everything is OK
4. Rerun step2 and it will suggest you changes by deleting the "merged" rule that does not exist

**Comment by bflad** [comment]
Hi @xakraz 👋  Sorry you're running into trouble here.

We made some improvements to `aws_security_group` handling in #4416 (released in [version 1.19.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1190-may-16-2018)).

Does upgrading your provider version help this issue?

**Comment by xakraz** [comment]
Hi @bflad , I will try it tomorrow and give you my feed back.

**Comment by mildwonkey** [comment]
@xakraz any luck? 

We are working on a [Community Gardening](https://www.terraform.io/docs/providers/aws/events/gardening-summer-2018.html) ticket cleanup - I don't want to close this if you are still seeing an issue.

**Comment by xakraz** [comment]
Hi @mildwonkey , sorry for the delay, I will check it at least on monday and answer tot this post.

Thank you so much for the effort and for the time :)

**Comment by jasonkuehl** [comment]
I just had this issue today. I can confirm that after removing the comment from the AWS console on the specific  rule I was able to apply.

* aws_security_group.partner_redshift: 1 error(s) occurred:

* aws_security_group.partner_redshift: Error revoking security group ingress rules: InvalidPermission.NotFound: The specified rule does not exist in this security group.

**Comment by bflad** [comment]
The specific issue above with `aws_security_group` rule descriptions should be resolved with version 1.19.0 of the AWS provider. If someone is still having trouble with the `aws_security_group` resource with newer versions of the AWS provider, please open a new issue filling out all the relevant issues of the issue template so we can further troubleshoot.
